huaweicloud/services/dms/resource_huaweicloud_dms_kafka_smart_connect_task.go

```diff
@@ -0,0 +1,382 @@
+				}
+				return rst
+			}
+		}
+	}
+	return nil
+}
+
+func resourceDmsKafkaSmartConnectTaskRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getKafkaSmartConnectTask: query DMS kafka smart connect task
+	var (
+		getKafkaSmartConnectTaskHttpUrl = "v2/{project_id}/connectors/{connector_id}/sink-tasks/{task_id}"
+		getKafkaSmartConnectTaskProduct = "dms"
+	)
+	getKafkaSmartConnectTaskClient, err := cfg.NewServiceClient(getKafkaSmartConnectTaskProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DMS Client: %s", err)
+	}
+
+	connectorID := d.Get("connector_id").(string)
+	getKafkaSmartConnectTaskPath := getKafkaSmartConnectTaskClient.Endpoint + getKafkaSmartConnectTaskHttpUrl
+	getKafkaSmartConnectTaskPath = strings.ReplaceAll(getKafkaSmartConnectTaskPath, "{project_id}",
+		getKafkaSmartConnectTaskClient.ProjectID)
+	getKafkaSmartConnectTaskPath = strings.ReplaceAll(getKafkaSmartConnectTaskPath, "{connector_id}", connectorID)
+	getKafkaSmartConnectTaskPath = strings.ReplaceAll(getKafkaSmartConnectTaskPath, "{task_id}", d.Id())
+
+	getKafkaSmartConnectTaskOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	getKafkaSmartConnectTaskResp, err := getKafkaSmartConnectTaskClient.Request("GET", getKafkaSmartConnectTaskPath,
+		&getKafkaSmartConnectTaskOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DMS kafka smart connect task")
+	}
+
+	getKafkaSmartConnectTaskRespBody, err := utils.FlattenResponse(getKafkaSmartConnectTaskResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("connector_id", connectorID),
+		d.Set("task_name", utils.PathSearch("task_name", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("destination_type", utils.PathSearch("destination_type", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("created_at", utils.FormatTimeStampUTC(int64(utils.PathSearch("create_time", getKafkaSmartConnectTaskRespBody, 0).(float64)/1000))),
+		d.Set("status", utils.PathSearch("status", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("topics", utils.PathSearch("topics", getKafkaSmartConnectTaskRespBody, nil)),
+		d.Set("obs_destination_descriptor", flattenObsDestinationDescriptor(getKafkaSmartConnectTaskRespBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenObsDestinationDescriptor(getKafkaSmartConnectTaskRespBody interface{}) []map[string]interface{} {
+	if getKafkaSmartConnectTaskRespBody == nil {
+		return nil
+	}
+
+	obsDestinationDescriptorStruct := utils.PathSearch("obs_destination_descriptor", getKafkaSmartConnectTaskRespBody, nil)
+	if obsDestinationDescriptorStruct == nil {
+		return nil
+	}
+
+	obsDestinationDescriptorList := make([]map[string]interface{}, 1)
+	params := map[string]interface{}{
```

Comment:
this will make `access_key` and `secret_key` be nil after refreshing the resource and cause a changed plan.

Reply:
access_key and secret_key are not set in read function.